### ISSUE TITLE
Implementation of homemade obs_transforms with trainable parameters

### ISSUE THREAD

**Issue by CarlosSandoval03** [facebookresearch/habitat-lab#1705]
## Habitat-Lab versions
Habitat-Lab: v2.4

Hi, hope it is the appropriate place to post this question, and thanks in advance. 

I am working with pointgoal navigation using the predefined configurations. I am transforming the observations using homemade transformations, which are 2 neural networks (using torch).  I have the problem that **the grads and the biases are being updated, but the weights remain exactly the same**, hence the network is not learning. 

In the past, I have successfully used homemade transformations that do not require training, but **I would like to ask if there is something I might be missing in order to apply obs_transforms with trainable parameters**. I am aware that given the complexity of Habitat it is unlikely to get a straight-forward answer, however I would appreciate if you have some input regarding what might be causing the issue, or which files/functions I should put more attention to. For clarification, I would provide more context as follows:
- I took as baseline a working version using the v2.2 of Habitat-Lab, so I am adapting the code to the differences with my version (v2.4).
- I know that the networks that perform the transformations are called properly in due time, they indeed transform the images, but it simply does not get any better with training.
- I was careful to include the code to perform the updates for the agent, the rollouts, and the losses (among others) according to my new requirements.
- I am plotting the histograms of every learnable parameter of the networks, that is why I know that grads and losses are being (properly) updated, but not the weights.
- I already made sure that the optimizer is receiving the parameters of the actor_critic as well as the ones from my custom made networks (they all have require_grad set to true). 
- I even tried to also include my transformations in the configs when the functions _create_updater and _create_policy (in habitat_baselines.rl.ppo.ppo.py) are. 
- According to the grads, it is unlikely that the problem is related to vanishing or exploding gradients.

I would greatly appreciate any ideas of what might be causing the problem.
